pyk/src/pyk/kast.py

```diff
@@ -1633,6 +1637,48 @@ def _kdefinition_empty_config(_sort):
 
         return _kdefinition_empty_config(sort)
 
+    def init_config(self, sort: KSort) -> KInner:
+
+        config_var_map = KVariable('__###CONFIG_VAR_MAP###__')
+
+        def _remove_config_var_lookups(_kast: KInner) -> KInner:
+            if type(_kast) is KApply and _kast.label.name.startswith('project:') and len(_kast.args) == 1:
+                _term = _kast.args[0]
+                if type(_term) is KApply and _term.label == KLabel('Map:lookup') and _term.args[0] == config_var_map:
+                    _token_var = _term.args[1]
+                    if type(_token_var) is KToken and _token_var.sort == KSort('KConfigVar'):
+                        return KVariable(_token_var.token)
+            return _kast
+
+        init_prods = [prod for prod in self.syntax_productions if 'initializer' in prod.att]
+        _init_prod = [prod for prod in init_prods if prod.sort == sort]
+        if len(_init_prod) != 1:
+            raise ValueError(f'Did not find unique initializer for sort: {sort}')
+        init_prod = _init_prod[0]
```

Comment:
We use this idiom quite a lot, let's extract a function to `pyk.util`. Maybe something like this:

```python3
from typing import Iterable, TypeVar, cast

T = TypeVar('T')


def single(iterable: Iterable[T]) -> T:
    it = iter(iterable)
    sentinel = object()
    
    fst = next(it, sentinel)
    if fst is sentinel:
        raise ValueError('Expected a single element, found: none')
    fst = cast(T, fst)

    snd = next(it, sentinel)
    if snd is not sentinel:
        raise ValueError(f'Expected a single element, found: {fst}, {snd}, ...')

    return fst
```

Reply:
Nice! Could we have this take an error message too please.

Reply:
Good idea. Or perhaps even

```python3
message: Union[str, Callable[[Tuple[T, ...]], str], None] = None
```

where the callable receives either `()` or `(fst, snd)`.

Reply:
This sounds like a more extensive change, made an issue: https://github.com/runtimeverification/k/issues/2820